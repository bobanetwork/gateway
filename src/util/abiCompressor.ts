/**
 * Script to prepare the human readable abi from the json.
 * based on the functions used out of the abi in network service.
 *
 * to prepare abi execute like
 * ```sh
 *  node ./util/abiCompressor.ts
 * ```
 *
 *
 */

import { ethers } from 'ethers'
import * as fs from 'fs'

const smart_contract_abi_paths = [
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/messaging/L2StandardBridge.sol/L2StandardBridge.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/standards/L2StandardERC20.sol/L2StandardERC20.json',
  './node_modules/@bobanetwork/core_contracts/artifacts/contracts/L2/predeploys/OVM_GasPriceOracle.sol/OVM_GasPriceOracle.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DiscretionaryExitFee.sol/DiscretionaryExitFee.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L1LiquidityPool.sol/L1LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/Teleportation.sol/Teleportation.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/LP/L2LiquidityPool.sol/L2LiquidityPool.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/BobaFixedSavings.sol/BobaFixedSavings.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance-token/BOBA.sol/BOBA.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegate.sol/GovernorBravoDelegate.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/DAO/governance/GovernorBravoDelegator.sol/GovernorBravoDelegator.json',
  './node_modules/@bobanetwork/contracts/artifacts/contracts/L2BillingContract.sol/L2BillingContract.json',
  './src/deployment/contracts/L1ERC20.json',
  './src/deployment/contracts/NFTMonsterV2.json',
  './src/deployment/contracts/AuthenticatedFaucet.json',
  './src/deployment/contracts/WAGMIv0.json',
  './src/deployment/contracts/WAGMIv1.json',
  './src/deployment/contracts/crosschain/EthBridge.json',
  './src/deployment/contracts/crosschain/L2StandardERC20.json',
  './src/deployment/contracts/crosschain/LZEndpointMock.json',
]

const networkServicesStr = fs.readFileSync(
  './src/services/networkService.ts',
  'utf8'
)

const compress = () => {
  for (const path of smart_contract_abi_paths) {
    const rawData = fs.readFileSync(path, 'utf8')
    const abiObject = JSON.parse(rawData)
    const name = abiObject.contractName
    let abi = abiObject.abi
    abi = abi.filter((component) => {
      if (component['type'] === 'function' && component['name']) {
        if (networkServicesStr.includes(component['name'])) {
          return true
        }
        return false
      }
    })
    const abiInterface = new ethers.utils.Interface(abi)
    const humanReadableABI = abiInterface.format(ethers.utils.FormatTypes.full)
    const stringifiedHumanReadableABI = JSON.stringify(humanReadableABI)
    const result = `const ${name}ABI = ${stringifiedHumanReadableABI}\n export default ${name}ABI`
    fs.writeFile(`./src/services/abi/${name}.abi.ts`, result, () => {
      console.log(`wrote ${name}`)
    })
  }
}

;(() => {
  compress()
})()
