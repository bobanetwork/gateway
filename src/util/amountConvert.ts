/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

// we use BigNumber here for decimal support
import BigNumber from 'bignumber.js'
import { BigNumberish } from 'ethers'

export const logAmount = (
  amount: BigNumberish,
  power: number,
  truncate = 0
) => {
  // toString as some methods seem to supply BigNumber
  const x = new BigNumber(amount?.toString())
  const exp = new BigNumber(10).pow(power)

  const calculated = x.div(exp)

  if (truncate > 0) {
    return calculated.toFixed(truncate)
  } else {
    return calculated.toFixed()
  }
}

/*Takes a value such as 3.92 and converts it into
a BigNumber in wei

Duplicates

ethers.utils.parseUnits( valueString , decimalsOrUnitName )   =>   BigNumber
*/

export const powAmount = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

/* more clearly named version of this */
export const toWei_String = (amount: number, decimals: number) => {
  const x = new BigNumber(amount)
  const exp = new BigNumber(10).pow(decimals)

  const calculated = x.multipliedBy(exp)
  return calculated.toFixed(0)
}

const lookupMap = {
  ETH: { provider: 'ethereum' },
  BOBA: { provider: 'boba-network' },
  OLO: { provider: 'oolongswap' },
  OMG: { provider: 'omisego' },
  USDC: { provider: 'usd-coin' },
  BNB: { provider: 'binancecoin' },
  tBNB: { provider: 'binancecoin' },
}

export const amountToUsd = (amount: number, lookupPrice, token) => {
  const { symbol } = token
  if (!symbol) {
    return 0
  }
  const currencyInfo = lookupMap[symbol]
  if (!!currencyInfo && !!lookupPrice[currencyInfo.provider]) {
    return amount * lookupPrice[currencyInfo.provider].usd
  } else if (!!lookupPrice[symbol.toLowerCase()]) {
    return amount * lookupPrice[symbol.toLowerCase()].usd
  } else {
    return 0
  }
}

export const formatLargeNumber = (num: number) => {
  const exp = Math.floor(Math.log10(Math.abs(num)) / 3)
  const scaledNum = Math.abs(num) / Math.pow(10, exp * 3)
  const formattedNumber = scaledNum.toFixed(2)
  const exponent = ['', 'k', 'M', 'B', 'T'][exp] || ''
  return isNaN(Number(formattedNumber)) ? '0' : formattedNumber + exponent
}
