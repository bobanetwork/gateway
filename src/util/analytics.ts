import ReactGA from 'react-ga4'
import { GA4_MEASUREMENT_ID } from './constant'

export const init = () => {
  if (window.location.hostname === 'localhost') {
    return null
  }
  if (GA4_MEASUREMENT_ID) {
    // init if the GA4 Measurement Id is available.
    ReactGA.initialize(GA4_MEASUREMENT_ID?.toString())
  }
}

export const sendPageView = (path: string) => {
  ReactGA.send({
    hitType: 'pageview',
    page: path,
    title: path,
  })
}

export const trackClick = (category: string, action: string, label: string) => {
  ReactGA.event({
    category,
    action,
    label,
  })
}

export default {
  init,
  sendPageView,
  trackClick,
}
