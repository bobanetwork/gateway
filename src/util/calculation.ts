/*
  Varna - A Privacy-Preserving Marketplace
  Varna uses Fully Homomorphic Encryption to make markets fair. 
  Copyright (C) 2021 Enya Inc. Palo Alto, CA

  This program is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation, either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

const countDecimals = (num) => {
  let len = 0
  try {
    num = Number(num)
    let str = num.toString().toUpperCase()
    if (str.split('E').length === 2) {
      // scientific notation
      let isDecimal = false
      if (str.split('.').length === 2) {
        str = str.split('.')[1]
        if (parseInt(str.split('E')[0], 10) !== 0) {
          isDecimal = true
        }
      }
      const x = str.split('E')
      if (isDecimal) {
        len = x[0].length
      }
      len -= parseInt(x[1], 10)
    } else if (str.split('.').length === 2) {
      // decimal
      if (parseInt(str.split('.')[1], 10) !== 0) {
        len = str.split('.')[1].length
      }
    }
  } catch (e) {
    throw e
  } finally {
    if (isNaN(len) || len < 0) {
      len = 0
    }
  }
  return len
}

export { countDecimals }
