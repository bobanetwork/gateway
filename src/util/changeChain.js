export const isChangingChain = JSON.parse(localStorage.getItem('changeChain'));
