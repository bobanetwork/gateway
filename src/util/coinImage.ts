import TESTLogo from 'assets/images/test.svg'

export const getCoinImage = (symbol: string): string => {
  const logoURIbase =
    'https://raw.githubusercontent.com/bobanetwork/token-list/main/assets'
  let url = `${logoURIbase}/${symbol?.toLowerCase()}.svg`

  if (symbol === 'test') {
    url = TESTLogo
  }
  return url
}
