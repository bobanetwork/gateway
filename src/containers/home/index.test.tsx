import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import CustomThemeProvider from 'themes'
import { mockedInitialState } from 'util/tests'
import Home from '.'

const mockStore = configureStore([thunk])

const renderHomeComponent = () => {
  return render(
    <Provider store={mockStore(mockedInitialState)}>
      <BrowserRouter>
        <CustomThemeProvider>
          <Home />
        </CustomThemeProvider>
      </BrowserRouter>
    </Provider>
  )
}

describe('Home ', () => {
  test('should match snapshot', () => {
    window.scrollTo = jest.fn()
    const { asFragment } = renderHomeComponent()
    expect(asFragment).toBeTruthy()
    expect(asFragment()).toMatchSnapshot()
  })
})
