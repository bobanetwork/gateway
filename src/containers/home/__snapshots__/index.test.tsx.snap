// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  should match snapshot 1`] = `
<DocumentFragment>
  .c12 {
  font-style: normal;
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 290px;
  -webkit-appearance: none;
  border-radius: 12px;
  cursor: pointer;
  border: 0px;
  outline: none;
  background: #AEDB01;
  padding: 16px 25px;
  font-weight: 700;
  font-size: 18px;
  box-shadow: 0px 2px 10px rgba(0,0,0,0.15);
  -webkit-transition: all 0.25s ease;
  transition: all 0.25s ease;
  min-width: auto;
  font-size: 14px;
  border-radius: 33px;
}

.c6:hover {
  background: #90B406;
}

.c1 {
  height: 73px;
  margin: 0px;
  padding: 0px 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-backdrop-filter: blur(7.5px);
  backdrop-filter: blur(7.5px);
  z-index: 9999;
  background: rgba(0,0,0,0.05);
}

.c2 {
  width: 30px;
  height: 32px;
  margin-right: 32px;
  background: url(boba-logo.png) no-repeat;
  background-position: 100%;
  background-size: contain;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 15px;
}

.c7 {
  display: none;
}

.c8 {
  stroke: #fff;
  fill: #fff;
}

.c9 {
  border-radius: 50%;
  cursor: pointer;
}

.c9:hover {
  background: #545454;
}

.c9 div {
  padding: 4px;
  height: 32px;
  width: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 24px;
}

.c4 {
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #A8A8A8;
}

.c4:hover,
.c4.active {
  color: #AEDB01;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 24px;
  border-radius: 33px;
  cursor: pointer;
  gap: 10px;
  font-size: 14px;
  font-style: normal;
  font-weight: 500;
  line-height: normal;
  color: #EEEEEE;
  border-radius: 28px;
  background: #393939;
  color: #5F5F5F;
  border: 2px solid #5F5F5F;
  background: transparent;
}

.c25:hover {
  color: #A8A8A8;
  border: 2px solid #AEDB01;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 24px 32px;
}

.c26 {
  border-color: #5F5F5F;
  box-sizing: border-box;
  width: 100%;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c28 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 32px;
}

.c29 {
  color: #5F5F5F;
  font-weight: 500;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 32px;
}

.c31 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  overflow: visible;
}

.c31 svg#docs path {
  fill: none;
  stroke: #5F5F5F;
}

.c31 path {
  fill: #5F5F5F;
}

.c31:hover svg#docs path {
  fill: none;
  stroke: #A8A8A8;
}

.c31:hover path {
  fill: #A8A8A8;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 24px;
}

.c24 {
  margin: 10px 0px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0;
  gap: 32px;
}

.c22 {
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #5F5F5F;
}

.c22:hover,
.c22.active {
  color: #A8A8A8;
}

.c23 {
  font-size: 14px;
  line-height: 16px;
  font-weight: 500;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #5F5F5F;
}

.c23:hover,
.c23.active {
  color: #A8A8A8;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
}

.c17 {
  min-height: calc(100vh - 73px);
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100vw;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  gap: 2px;
}

.c11 {
  position: relative;
  -webkit-transition: max-height 0.4s;
  transition: max-height 0.4s;
  padding: 5px 10px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 10px;
  width: 100%;
  background: #AEDB01;
  color: #000000;
}

.c11.open {
  max-height: 60px;
}

.c14 {
  cursor: pointer;
}

.c15 {
  cursor: pointer;
  border-radius: 50%;
}

.c15:hover {
  background: #CEE967;
}

.c15 div {
  padding: 1px;
  height: 24px;
  width: 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
}

.c13 {
  text-align: center;
  max-width: 1440px;
  margin: 0 auto;
  padding: 10px 0;
}

.c13 a {
  color: #000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  opacity: 0.65;
  text-transform: capitalize;
}

.c16 {
  height: 12px;
  width: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c16 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c16 svg {
  max-width: 24px;
  min-width: 10px;
  height: auto;
  stroke: #000;
}

@media (max-width:767px) {
  .c1 {
    padding: 0px 16px;
  }
}

@media (max-width:767px) {
  .c2 {
    width: 32px;
    height: 32px;
    margin-right: 12px;
  }
}

@media (max-width:767px) {
  .c5 {
    gap: 8px;
  }
}

@media (max-width:767px) {
  .c7 {
    display: block;
  }
}

@media (max-width:767px) {
  .c8 {
    display: block;
    height: 34px;
    width: 32px;
  }

  .c8 div {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:767px) {
  .c9 {
    display: none;
  }
}

@media (max-width:767px) {
  .c3 {
    display: none;
  }
}

@media (max-width:767px) {
  .c26 {
    display: none;
  }
}

@media (max-width:767px) {
  .c27 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    gap: 24px;
  }
}

@media (max-width:767px) {
  .c20 {
    width: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 24px;
  }
}

@media (max-width :980px) {
  .c11 {
    padding: 0 10px;
  }

  .c11.open.expand {
    max-height: 120px;
  }
}

@media (max-width :767px) {
  .c13 {
    padding: 5px 10px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
      id="header"
    >
      <a
        class=""
        href="/bridge"
      >
        <div
          class="c2"
        />
      </a>
      <div
        class="c3"
      >
        <a
          class="c4 ({
          isActive
        }) => isActive ? 'active' : ''"
          href="/bridge"
        >
          Bridge
        </a>
        <a
          class="c4 ({
          isActive
        }) => isActive ? 'active' : ''"
          href="/history"
        >
          History
        </a>
        <a
          class="c4 ({
          isActive
        }) => isActive ? 'active' : ''"
          href="/stake"
        >
          Stake
        </a>
        <a
          class="c4 ({
          isActive
        }) => isActive ? 'active' : ''"
          href="/dao"
        >
          Dao
        </a>
      </div>
      <div
        class="c5"
      >
        <button
          class="c6"
          label="Connect Wallet"
          style="white-space: nowrap;"
          type="button"
        >
           Connect Wallet
        </button>
        <div
          class="c7"
        >
          <div
            class="c8"
          >
            <div>
              <div
                data-src="hamburger.svg"
                src=""
              />
            </div>
          </div>
        </div>
        <div
          class="c9"
          title="light-icon"
        >
          <div>
            <div>
              <div
                data-src="theme-sun.svg"
                src=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c10"
    >
      <div
        class="c11 warning"
        data-testid="banner-item"
      >
        <div
          class=""
        >
          <p
            class="c12 c13"
          >
             Boba Light bridge has been released and earn is sunset, to withdraw funds from liquidity 
            <a
              href="/earn"
            >
              CLICK HERE
            </a>
             
          </p>
        </div>
        <div
          class="c14"
        >
          <div
            class="c15"
            data-testid="close-icon-earn-banner-deprecation"
          >
            <div
              class="c16"
            >
              <div>
                <div
                  data-src="close.svg"
                  src=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c17"
    />
    <div
      class="c18"
    >
      <div
        class="c19"
      >
        <div
          class="c20"
        >
          <div
            class="c21"
            id="footerLinks"
          >
            <a
              class="c22"
              href="https://docs.boba.network/faq"
              target="_blank"
            >
              FAQs
            </a>
            <a
              class="c22"
              href="https://docs.boba.network/for-developers"
              target="_blank"
            >
              Dev Tools
            </a>
            <a
              class="c23"
              href="/bobascope"
            >
              Bobascope
            </a>
            <a
              class="c22"
              href="https://boba.network"
              target="_blank"
            >
              Boba Network Website
            </a>
            <a
              class="c22"
              href="https://boba.network/terms-of-use/"
              target="_blank"
            >
              Terms of Use
            </a>
          </div>
          <div
            class="c24"
          >
            <div
              class="c25"
            >
              Block explorers
            </div>
          </div>
        </div>
      </div>
      <hr
        class="c26"
      />
      <div
        class="c27"
        id="socialLinks"
      >
        <div
          class="c28"
        >
          <p
            class="c12 c29"
          >
            ©2023 Enya Labs
          </p>
          <p
            class="c12 c29"
          >
            v
          </p>
        </div>
        <div
          class="c30"
        >
          <a
            aria-label="bobadocs"
            class="c31"
            href="https://docs.boba.network"
            target="_blank"
          >
            <svg
              fill="none"
              height="22"
              id="docs"
              viewBox="0 0 18 22"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.5 1H3C2.46957 1 1.96086 1.21071 1.58579 1.58579C1.21071 1.96086 1 2.46957 1 3V19C1 19.5304 1.21071 20.0391 1.58579 20.4142C1.96086 20.7893 2.46957 21 3 21H15C15.5304 21 16.0391 20.7893 16.4142 20.4142C16.7893 20.0391 17 19.5304 17 19V6.5L11.5 1Z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M11 1V7H17"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M13 12H5"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M13 16H5"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
              <path
                d="M7 8H5"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </a>
          <a
            aria-label="twitter"
            class="c31"
            href="https://boba.eco/twitter"
            target="_blank"
          >
            <svg
              height="22"
              viewBox="0 0 24 22"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.5337 5.92715C21.5435 6.14946 21.5435 6.36168 21.5435 6.58399C21.5532 13.3242 16.6011 21.1053 7.54509 21.1053C4.87409 21.1053 2.25183 20.307 0 18.8114C0.389927 18.8619 0.779854 18.8821 1.16978 18.8821C3.38262 18.8821 5.53696 18.1141 7.28188 16.6893C5.17628 16.6489 3.32413 15.224 2.68075 13.1423C3.42161 13.2939 4.18197 13.2636 4.90333 13.0514C2.61251 12.5865 0.965069 10.4947 0.955321 8.05937C0.955321 8.03916 0.955321 8.01894 0.955321 7.99873C1.63769 8.39284 2.4078 8.61516 3.18765 8.63537C1.03331 7.13978 0.360682 4.15872 1.66694 1.8244C4.17222 5.01767 7.85703 6.94778 11.8148 7.15999C11.4151 5.39157 11.961 3.53219 13.238 2.27914C15.2169 0.349025 18.3363 0.450078 20.208 2.50145C21.3095 2.27914 22.3721 1.85471 23.3371 1.2585C22.9667 2.44082 22.1966 3.44125 21.173 4.07788C22.1478 3.95662 23.1032 3.68377 24 3.27956C23.3371 4.3103 22.4988 5.19957 21.5337 5.92715Z"
              />
            </svg>
          </a>
          <a
            aria-label="discord"
            class="c31"
            href="https://boba.eco/discord"
            target="_blank"
          >
            <svg
              fill="none"
              height="19"
              viewBox="0 0 24 19"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.2415 2.18563C18.7257 1.49012 17.1003 0.9777 15.4007 0.684215C15.3698 0.67855 15.3389 0.692705 15.3229 0.721016C15.1139 1.09283 14.8823 1.57788 14.7202 1.95913C12.8922 1.68547 11.0736 1.68547 9.28318 1.95913C9.12099 1.56941 8.88103 1.09283 8.67104 0.721016C8.6551 0.69365 8.62418 0.679495 8.59323 0.684215C6.89465 0.976762 5.26918 1.48918 3.75245 2.18563C3.73932 2.19129 3.72807 2.20074 3.7206 2.213C0.637438 6.81917 -0.207169 11.3121 0.207167 15.7494C0.209041 15.7711 0.221228 15.7919 0.238101 15.8051C2.27228 17.2989 4.24273 18.2058 6.17659 18.8069C6.20754 18.8164 6.24034 18.8051 6.26003 18.7796C6.71749 18.1549 7.12527 17.4962 7.4749 16.8035C7.49553 16.7629 7.47584 16.7148 7.43366 16.6987C6.78686 16.4534 6.17097 16.1542 5.57852 15.8145C5.53166 15.7871 5.52791 15.7201 5.57102 15.688C5.69569 15.5946 5.8204 15.4974 5.93944 15.3992C5.96098 15.3813 5.99099 15.3775 6.01631 15.3889C9.90842 17.1659 14.1221 17.1659 17.9683 15.3889C17.9936 15.3766 18.0236 15.3804 18.0461 15.3983C18.1652 15.4965 18.2898 15.5946 18.4154 15.688C18.4585 15.7201 18.4557 15.7871 18.4089 15.8145C17.8164 16.1608 17.2005 16.4534 16.5528 16.6978C16.5106 16.7138 16.4919 16.7629 16.5125 16.8035C16.8696 17.4952 17.2774 18.1539 17.7264 18.7786C17.7452 18.8051 17.7789 18.8164 17.8099 18.8069C19.7531 18.2058 21.7235 17.2989 23.7577 15.8051C23.7755 15.7919 23.7868 15.772 23.7887 15.7503C24.2845 10.6204 22.9581 6.16426 20.2724 2.21394C20.2659 2.20074 20.2546 2.19129 20.2415 2.18563ZM8.05612 13.0476C6.88433 13.0476 5.91881 11.9718 5.91881 10.6506C5.91881 9.3294 6.86561 8.25361 8.05612 8.25361C9.25598 8.25361 10.2122 9.33884 10.1934 10.6506C10.1934 11.9718 9.2466 13.0476 8.05612 13.0476ZM15.9585 13.0476C14.7867 13.0476 13.8212 11.9718 13.8212 10.6506C13.8212 9.3294 14.768 8.25361 15.9585 8.25361C17.1584 8.25361 18.1145 9.33884 18.0958 10.6506C18.0958 11.9718 17.1584 13.0476 15.9585 13.0476Z"
                fill="#5F5F5F"
              />
            </svg>
          </a>
          <a
            aria-label="telegram"
            class="c31"
            href="https://boba.eco/telegram"
            target="_blank"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M2.03714 11.2933C8.23863 8.59144 12.3739 6.81019 14.443 5.94958C20.3507 3.49236 21.5783 3.06551 22.3784 3.05142C22.5544 3.04832 22.9479 3.09193 23.2028 3.29875C23.418 3.47337 23.4772 3.70927 23.5055 3.87484C23.5339 4.04041 23.5691 4.41758 23.5411 4.71228C23.2209 8.07603 21.8357 16.2389 21.131 20.0064C20.8328 21.6005 20.2456 22.1351 19.6772 22.1874C18.4418 22.301 17.5037 21.371 16.3072 20.5866C14.4349 19.3593 13.3772 18.5953 11.5598 17.3977C9.45952 16.0136 10.8211 15.2529 12.018 14.0097C12.3313 13.6844 17.7743 8.73353 17.8796 8.28441C17.8928 8.22824 17.905 8.01887 17.7806 7.90831C17.6563 7.79775 17.4727 7.83556 17.3402 7.86563C17.1524 7.90825 14.1614 9.88519 8.36716 13.7965C7.51817 14.3794 6.74918 14.6635 6.0602 14.6486C5.30064 14.6322 3.83957 14.2191 2.75341 13.8661C1.42119 13.433 0.362372 13.2041 0.454571 12.4686C0.502595 12.0855 1.03012 11.6938 2.03714 11.2933Z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
